src/lib/parsers/sol-json.c

```diff
@@ -289,28 +289,17 @@ sol_json_token_get_double(const struct sol_json_token *token, double *value)
      * of bytes.
      */
 
-    *value = sol_util_strtodn(token->start, &endptr,
+    errno = 0;
+    tmp_value = sol_util_strtodn(token->start, &endptr,
         sol_json_token_get_size(token), false);
+    SOL_INT_CHECK(errno, != 0, -errno);
```

Comment:
OK, after some discussion we decided to keep the code as it is now and handle errors in module side.
The main point of divergence was that inf and nan are acceptable values for a float, but shouldn't be passed via JSON.